### PR TITLE
Feature/235 custom language

### PR DESCRIPTION
**Pull request by aouame**
I implemented language selection follwing device's local language.
Contents are now retreived and searched for on search fragment given local language with appropriate translated blockswrapper.
I added a new english content for the demo visit on firebase to test it.

cap1
cap2
cap3
cap4

### CODE REVIEW

**Comment by aouame** [comment]
@lucasrlt I refactored as you told me. I thought myself it was related to the content but it works this way.

**Comment by aouame** [comment]
@lucasrlt I refactored as you told me. I thought it was related to the context but it works this way.

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -1,6 +1,22 @@
 <component name="ProjectCodeStyleConfiguration">
```
Comment (Mortadha-abderrhim) [inline .idea/codeStyles/Project.xml:1]: I think You should Roll back the changes to this file and wise move adding .idea to gitignore  as it's really starting to be an issue

File: app/src/main/java/com/pindex/main/adapters/FullExperienceAdapter.kt [hunk 2]
```diff
@@ -24,7 +26,7 @@ class FullExperienceAdapter :
             // Experience type: activity, event or tour
             experienceType.text = "Pindex ${experience.type?.capitalize()}"
 
-            val content: ContentDto? = experience.content?.get(0)
+            val content = ActivityUtils.getContentFromLanguage(experience) // default language is french
```
Comment (Mortadha-abderrhim) [inline app/src/main/java/com/pindex/main/adapters/FullExperienceAdapter.kt:29]: I think it is not a good idea to add this comment here, just add it to the getContentFromLanguage doc

File: app/src/main/java/com/pindex/main/adapters/NearbyExperienceAdapter.kt [hunk 3]
```diff
@@ -27,7 +29,8 @@ class NearbyExperienceAdapter :
          * Bind the given [experience] data to the view.
          */
         override fun bind(experience: ExperienceDto) {
-            val content: ContentDto? = experience.content?.get(0)
+            val content =
+                ActivityUtils.getContentFromLanguage(experience) // default language is french
```
Comment (Mortadha-abderrhim) [inline app/src/main/java/com/pindex/main/adapters/NearbyExperienceAdapter.kt:33]: same

File: app/src/main/java/com/pindex/main/ui/activities/ExperienceActivity.kt [hunk 4]
```diff
@@ -80,8 +82,12 @@ class ExperienceActivity : AppCompatActivity() {
         viewModel.experience.value = experience
 
         // Get the experience Pindex blocks
-        val blocks: ArrayList<BlockDto>? = experience?.content?.get(0)?.blocksWrapper?.blocks
+        val content =
+            experience?.let { ActivityUtils.getContentFromLanguage(it) }  // default language is french
```
Comment (Mortadha-abderrhim) [inline app/src/main/java/com/pindex/main/ui/activities/ExperienceActivity.kt:86]: here too

File: app/src/main/java/com/pindex/main/utils/ActivityUtils.kt [hunk 5]
```diff
@@ -18,4 +21,14 @@ object ActivityUtils {
         }
         return false
     }
+
+    /**
+     * retreives content based on device's language
```
Comment (Mortadha-abderrhim) [inline app/src/main/java/com/pindex/main/utils/ActivityUtils.kt:26]: It goes right here I think

File: app/src/main/java/com/pindex/main/viewmodels/SearchViewModel.kt [hunk 6]
```diff
@@ -39,8 +41,10 @@ class SearchViewModel : ViewModel() {
 
     private val dataSourceFactory = SearcherSingleIndexDataSource.Factory(searcher) { hit ->
         val exp = JsonUtils.experienceDtoFromJson(hit.json)
-        if (exp.content != null) SearchResult(exp.content?.get(0)?.title.toString(), exp)
-        else SearchResult("TEST EXPERIENCE", JsonUtils.failedJson)
+        if (exp.content != null) {
+            val content = ActivityUtils.getContentFromLanguage(exp)  // default language is french
```
Comment (Mortadha-abderrhim) [inline app/src/main/java/com/pindex/main/viewmodels/SearchViewModel.kt:45]: again

**Review by Mortadha-abderrhim — CHANGES_REQUESTED** [review]

**Review by Mortadha-abderrhim — APPROVED** [review]
GJ, clean and to the point.

**Review by hadiyahya11 — APPROVED** [review]
Nice PR.Good job reformating the code.

**Pull request merged**